NorthwindApp/Northwind.Web.Tests/ApplienceTesting.cs

```diff
@@ -0,0 +1,311 @@
+﻿using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Northwind.Model;
+using Microsoft.Extensions.DependencyInjection;
+using AngleSharp;
+using AngleSharp.Dom;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+using AngleSharp.Html.Parser;
+using System.Security.Cryptography;
+using System.Xml.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using AngleSharp.Html.Dom;
+
+namespace Northwind.Web.Tests
+{
+    [TestClass]
+    public class ApplienceTesting
+    {
+        [TestMethod]
+        public async Task Home_ReturnViewResult_With_Links()
+        {
+            var client = new HttpClient() { BaseAddress = new Uri("http://localhost:5000") };
+
+            var response = await client.GetStringAsync("/");
+            var document = GetDocument(response);
+
+            // var links = document.Links.OfType<IHtmlAnchorElement>()
+            //     .Select(l => l.Href).Where(l => l.Contains("Categories"));
+
+            var links = document.Links.OfType<IHtmlAnchorElement>()
+                .Select(l => l.Href);
+
+            links.Any(l=>l.EndsWith("/")).Should().BeTrue();
+            links.Any(l=>l.Contains("Categories")).Should().BeTrue();
```

Comment:
Я бы всё же такие проверки заменил на 
```suggestion
            links.Where(l=>l.EndsWith("/")).Should().BeNotEmpty();
            links.Where(l=>l.Contains("Categories")).Should().BeNotEmpty();
```
По смыслу - то же самое, но разница будет в генерируемых ошибках. И в случае с моим вариантом, имхо, будет более понятная ошибка про то, что коллекция пустая. 
Впрочем, здесь вообще, имхо, напрашивается кастомное сообщение об ошибке.

Reply:
Исправил. Также добавил сохранение ссылок в список там, где в последствии обращение к ним происходит несколько раз.